### PR TITLE
Add XGBoost and LightGBM examples

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Adds XGBoost and LightGBM examples.

## How is this patch tested?

Manually

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: examples/lightgbm/conda.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: lightgbm-example
+channels:
+  - defaults
+  - conda-forge
+dependencies:
+  - python=3.6
+  - lightgbm
+  - pip:
+      - git+https://github.com/mlflow/mlflow
```
Comment (harupy) [inline examples/lightgbm/conda.yaml]: The current latest version (1.4.0) does not yet support the XGBoost and LightGBM flavors (the next version will). Should I wait for the next release?

Comment (smurching) [reply examples/lightgbm/conda.yaml]: Good question, yeah we plan to release 1.5 some time next week, so maybe we should update this to mlflow==1.5 then.

**Review by smurching — COMMENTED** [review]
Thanks for the PR, mostly looks good! Would you mind also updating tests/examples/test_examples.py to run these example scripts? It'd also be nice to list the new examples under https://github.com/mlflow/mlflow/blob/master/docs/source/tutorials-and-examples/index.rst.

File: examples/lightgbm/conda.yaml (hunk 1 above)
Comment (harupy) [reply examples/lightgbm/conda.yaml]: ok, I'll fix this.

**Comment by harupy** [comment]
@smurching 
Ok, I'll update them.

**Comment by harupy** [comment]
@smurching updated the doc and test.

**Review by smurching — APPROVED** [review]
LGTM thanks @harupy!

**Pull request merged**